### PR TITLE
add CLI parameter to define a fixed port for dynamodb

### LINKED ISSUE #21: dynamoDB / direct access
How can I use the same dynamodb which is created/used by AppSync Emulator to write/read from a normal lambda function called by serverless-offline?

On which local port can I directly access the dynamodb?

Thanks
Andreas
ps. Thanks for creating the AppSync emulator - great work!

### PR DESCRIPTION
**Pull request by aheissenberger**
This commit resolves #21 and adds a new cli parameter to define a fixed port:
`--dynamodb-port <portnumber>`

### CODE REVIEW

File: packages/appsync-emulator-serverless/bin/server.js [hunk 1]
```diff
@@ -34,14 +34,21 @@ const main = async () => {
...
-  let { port, path: serverlessPath } = parser.parseArgs();
+  parser.addArgument(['--dynamodb-port'], {
+    help: 'Port to bind the dynamodb to (default is any free port)',
+    type: 'int',
+  });
+
+  let { port, path: serverlessPath, dynamodb_port } = parser.parseArgs();
   port = port || 0;
   serverlessPath = serverlessPath || process.cwd();
+  dynamodb_port = dynamodb_port || null;
```
Comment (lightsofapollo) [inline packages/appsync-emulator-serverless/bin/server.js]: nit: `dynamodbPort`

**Comment by lightsofapollo** [comment]
@aheissenberger Looks good aside from the linter/prettier comments. Please fix those and we will happily land and release a new version! Thank you for your contribution 🎉

File: packages/appsync-emulator-serverless/bin/server.js [hunk 2]
```diff
@@ -54,6 +61,15 @@ const main = async () => {
   const serverless = path.join(path.dirname(pkgPath), 'serverless.yml');
   const server = await createServer({ serverless, port, dynamodb });
   console.log('started at url:', server.url);
+  dynamodb_port && console.log(
```
Comment (cbaron) [inline packages/appsync-emulator-serverless/bin/server.js]: I'm not sure if its worthwhile to print all of this.  Perhaps we can use what's in #23 ?

File: packages/appsync-emulator-serverless/README.md [hunk 3]
```diff
@@ -25,6 +25,24 @@ This package will download and run the dynamodb emulator as part of it's appsync
...
 yarn appsync-emulator --port 62222
 ```
+#### dynamodb with fixed port
+
+optional start dynamodb at a fixed port - e.g. 8000
+```sh
+# NOTE unless you assign a specific port for dynamodb a random one will be chosen.
+yarn appsync-emulator --port 62222 --dynamodb-port 8000
+```
+to access the dynamodb instance you need to use the following configuration:
```
Comment (cbaron) [inline packages/appsync-emulator-serverless/README.md]: Do you mind changing this to ?

`to access the dynamodb instance using javascript`..

and including 

`const { DynamoDB } = require('aws-sdk');`

?

**Comment by newswim** [comment]
👋 @aheissenberger, if you want to get to view the current lint-related issues (which is causing the Travis build to fail), you can run:

```
yarn lint
```

Unfortunately you will still have to manually make the changes since some involve undeclared variable names :-\

File: packages/appsync-emulator-serverless/bin/server.js [hunk 4]
```diff
@@ -54,6 +61,15 @@ const main = async () => {
...
+    `dynamodb config:
+    {
+      endpoint: 'http://localhost:${dynamodb_port}',
```
Comment (newswim) [inline packages/appsync-emulator-serverless/bin/server.js]: forgot to rename variable here.

File: packages/appsync-emulator-serverless/bin/server.js [hunk 5]
```diff
@@ -63,15 +63,16 @@ const main = async () => {
   const serverless = path.join(path.dirname(pkgPath), 'serverless.yml');
   const server = await createServer({ serverless, port, dynamodb });
   console.log('started at url:', server.url);
-  dynamodb_port && console.log(
-    `dynamodb config:
+  if (dynamodbPort)
```
Comment (newswim) [inline packages/appsync-emulator-serverless/bin/server.js]: add `/* eslint-disable no-console */` before line 65 and it should build. Sorry for all the tedium!

Comment (aheissenberger) [reply packages/appsync-emulator-serverless/bin/server.js]: no problem - all my fault ;-)

**Pull request merged**